dp/generate_climos.py

```diff
@@ -39,6 +39,86 @@
 cdo = Cdo()
 
 
+def has_climo_periods(input_file, climo):
+    '''
+    This function checks if the given input NetCDF dataset has a variable "climo_periods".
+    generate_climos requires the input dataset to have the variable, thus this function will
+    warn if there is no such variable.
+    '''
+    periods = input_file.climo_periods.keys() & climo
+    logger.info("climo_periods: {}".format(periods))
+    if len(periods) == 0:
+        logger.info(f"{input_file.filepath()} has no variable 'climo_periods'")
+    return periods
+
+
+def is_NetCDF(filepath):
+    '''
+    This function checks if the given input file path is a vaild NetCDF file. It returns
+    NetCDF dataset of the input file. Ohterwise, it halts the program.
+    '''
+    try:
+        return CFDataset(filepath)
+    except Exception as e:
```

Comment:
Good idea to catch this exception and print a log message. However, a few coding points here:
- It is better not to catch a generic `Exception` -- much too broad. Find out what exception is raised  when a non-NetCDF4 file is attempted to be opened, and catch that.
- It should not, generally speaking, be the business of a method called several levels deep to call `sys.exit()`. That should be the business of the highest levels of the script. If the code in question is going to be orchestrated by some higher level function or process (e.g. for parallelization, or some other purpose), `sys.exit()` makes it impossible for the coordinator to do its job.
- This is not an `info` level error: It is a serious error that means the script cannot do anything at all, and should be logged as such. Consider `log.exception()` or `log.critical()` here.

Reply:
Also see note below as to why we probably don't even want or need this function.